x-pack/libbeat/common/aws/credentials.go

```diff
@@ -143,17 +155,42 @@ func getRoleArn(config ConfigAWS, awsConfig awssdk.Config) awssdk.Config {
 // EnrichAWSConfigWithEndpoint function enabled endpoint resolver for AWS
 // service clients when endpoint is given in config.
 func EnrichAWSConfigWithEndpoint(endpoint string, serviceName string, regionName string, awsConfig awssdk.Config) awssdk.Config {
+	var eurl string
 	if endpoint != "" {
 		parsedEndpoint, _ := url.Parse(endpoint)
 		if parsedEndpoint.Scheme != "" {
 			awsConfig.EndpointResolver = awssdk.ResolveWithEndpointURL(endpoint)
 		} else {
 			if regionName == "" {
-				awsConfig.EndpointResolver = awssdk.ResolveWithEndpointURL("https://" + serviceName + "." + endpoint)
+				eurl = "https://" + serviceName + "." + endpoint
 			} else {
-				awsConfig.EndpointResolver = awssdk.ResolveWithEndpointURL("https://" + serviceName + "." + regionName + "." + endpoint)
+				eurl = "https://" + serviceName + "." + regionName + "." + endpoint
 			}
+			awsConfig.EndpointResolver = awssdk.ResolveWithEndpointURL(eurl)
 		}
 	}
 	return awsConfig
 }
+
+//Create AWS service name based on Region and FIPS
+func CreateServiceName(serviceName string, fipsEnabled bool, region string) string {
+	if fipsEnabled {
+		OptionalGovCloudFIPS := []string{"s3"}
+		_, found := Find(OptionalGovCloudFIPS, serviceName)
```

Comment:
Minor nit.  If you switch to a Map, you can get rid of the `Find` function.  Also might be nice to have the `OptionalGovCloudFIPS` var be package level with a godoc comment.

Reply:
I can change it to a map.

Reply:
as for where to put the variable, i'm not a GoLang expert to know what is best but I can't see the variable being used outside of this function.

Reply:
I added the map

Reply:
Reasoning behind moving variable to top level is for [doc comments](https://go.dev/doc/effective_go#commentary)  That way the comment shows up in the godoc output.  Again not a big deal, and I'm ok with merging without moving.

Reply:
I think i moved it correctly.  Please let me know otherwise.